tests/e2e/e2e_suite_test.go

```diff
@@ -58,224 +53,285 @@ const (
-}
+			_, _ = fmt.Fprintf(GinkgoWriter, "Helm chart installation: %q completed.\n",
+				testContext.chart)
+			_, _ = fmt.Fprintf(GinkgoWriter, "Helm chart installation: new %q release name.\n",
+				helmReleaseName)
+		})
 
-func (s *Suite) TestDCGMExporter() {
-	s.DCGMExporterPrechecks()
+		labelMap := map[string]string{dcgmExporterPodNameLabel: dcgmExporterPodNameLabelValue}
 
-	s.Run("Create workload pod", func() {
+		It("should create dcgm-exporter pod", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Pod creation verification: started")
 
-		s.T().Log("Creating a workload pod...")
+			Eventually(func(ctx context.Context) bool {
+				pods, err := kubeClient.GetPodsByLabel(ctx, testContext.namespace, labelMap)
+				if err != nil {
+					Fail(fmt.Sprintf("Pod creation: Failed with error: %v", err))
+					return false
+				}
 
-		var err error
+				return len(pods) == 1
+			}).WithPolling(time.Second).Within(15 * time.Minute).WithContext(ctx).Should(BeTrue())
 
-		s.workloadPod, err = s.k8SClient.CreatePod(s.ctx,
-			s.namespace,
-			s.labels,
-			workloadPodName,
-			workloadContainerName,
-			workloadImage,
-		)
+			_, _ = fmt.Fprintln(GinkgoWriter, "Pod creation verification: completed")
+		})
 
-		s.Require().NoError(err, "Cannot create workload pod")
+		It("should read pod", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Read dcgm-exporter pod: started")
+			pods, err := kubeClient.GetPodsByLabel(ctx, testContext.namespace, labelMap)
+			if err != nil {
+				Fail(fmt.Sprintf("Pod creation: Failed with error: %v", err))
+			}
 
-		s.Require().Eventuallyf(func() bool {
-			isReady, err := s.k8SClient.CheckPodCondition(s.ctx, s.namespace, s.workloadPod.Name, corev1.PodScheduled)
-			s.Require().NoErrorf(err, "Cannot get pod status: %v", err)
-			return isReady
-		}, 15*time.Minute, time.Second, "Failed to create pod: %s", s.workloadPod.Name)
+			Expect(pods).Should(HaveLen(1))
+
+			dcgmExpPod = &pods[0]
+
+			_, _ = fmt.Fprintln(GinkgoWriter, "Read dcgm-exporter pod: completed")
+		})
+
+		It("should ensure that the dcgm-exporter pod is ready", func(ctx context.Context) {
+			_, _ = fmt.Fprintln(GinkgoWriter, "Checking pod status: started")
+			Eventually(func(ctx context.Context) bool {
+				isReady, err := kubeClient.CheckPodStatus(ctx,
+					testContext.namespace,
+					dcgmExpPod.Name,
+					func(namespace, podName string, status corev1.PodStatus) (bool, error) {
+						for _, c := range status.Conditions {
+							if c.Type != corev1.PodReady {
+								continue
+							}
+							if c.Status == corev1.ConditionTrue {
+								return true, nil
+							}
+						}
 
-		s.T().Log("The workload was created.")
-	})
+						for _, c := range status.ContainerStatuses {
+							if c.State.Waiting != nil && c.State.Waiting.Reason == "CrashLoopBackOff" {
+								return false, fmt.Errorf("pod %s in namespace %s is in CrashLoopBackOff", podName, namespace)
+							}
+						}
+
+						return false, nil
+					})
```

Comment:
@rohit-arora-dev, The good of the "CrashLookBackOff" scenario is to run the dcgm-exporter with `dcp-metrics-included.csv` on a consumer-grade GPU instead of a data-center grade.  The metric DCGM_FI_PROF_GR_ENGINE_ACTIVE is not supported, so DCGM will fail and cause the  "CrashLookBackOff" on customer-grade GPU.